### PR TITLE
RavenDB-15482 Flatten nested fields under Settings when editing the D…

### PR DESCRIPTION
**Pull request by Danielle9897**
…atabase Record

### CODE REVIEW

**Comment by ml054** [comment]
Please make flatten code recursive.

File: src/Raven.Studio/typescript/viewmodels/database/settings/databaseRecord.ts [hunk 1]
```diff
@@ -160,6 +162,51 @@ class databaseRecord extends viewModelBase {
...
+        if (!settings) {
+            return dto;
+        }
+        
+        const settingsKeys = Object.keys(settings);
+        if (settingsKeys.length === 0) {
+            return dto;
+        }
+
+        let flattenedSettings = { fs: {} };
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/databaseRecord.ts]: I don't see the point to detect if we should flatten or not. Just check if settings are present if yes then call flatten and this is end of story. 

```
  function flattenObject(obj, parent, res = {}){
    for(let key in obj){
        let propName = parent ? parent + '.' + key : key;
        if(typeof obj[key] == 'object'){
            flattenObj(obj[key], propName, res);
        } else {
            res[propName] = obj[key];
        }
    }
    return res;
}
```

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/settings/databaseRecord.ts]: see: https://stackoverflow.com/questions/44134212/best-way-to-flatten-js-object-keys-and-values-to-a-single-depth-array

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/databaseRecord.ts]: done

**Review by ml054 — APPROVED** [review]

**Pull request merged**